### DISCUSSION TITLE
How can i insert \033 ansi color into the arguments?

### DISCUSSION THREAD

**Discussion by zacard-orc** [babel/babel#14731] [Q&A] [answered]
Hi , I want to insert some ansi color text into the arguments ?

eg:
```javascript
const hello = types.stringLiteral('\033[38;2;255;82;197mHello')
bbpath.node.arguments.unshift(hello)
```
after this code, i see types object.value was \033[38;2;255;82;197mHello
but after the transformation ,the output code will be \\033[38;2;255;82;197mHello, it's a \ char added

i debug the full flow , the official object has two props
```text
value => \\033[38;2;255;82;197mHello
rawValue => \033[38;2;255;82;197mHello
```

i try some parameters by the TransformOptions and GeneratorOptions, but it's not work

so, i pub this discussion to need help? thx ~😊

**Comment by nicolo-ribaudo** [comment]
I think some of the `\`s in your description are missing, but maybe this helps?
https://astexplorer.net/#/gist/969bbe6ab95b4aff5e67e53c144fa40c/f2f34a527289fb41d05f6f715f0507f4f15cd182

It's correct that the string in the output has `\\`, since in JavaScript you cannot have a lone `\` in strings but you must escape it. If you try coping the output code of that ASTExplorer link and run it in your console, you will see that the result has a single `\` when evaluated. You can also verify it like this:
```js
let str = "\\033[38;2;255;82;197mHello";
console.log(str[0]); // \ or \\, depending on how your console prints it
console.log(str[1]); // 0: if the second character is 0, there can be only a single \ before that!
```

**Comment by zacard-orc** [comment, marked as answer]
Along with your advice. I try other params combination, and it worked! 😊 when i pass \x1b other than \033, 
thx a lot @nicolo-ribaudo 🤝

```
generatorOpts: {
            jsescOption: {
                minimal: false,
                compact: true,
            },
        },
```
```javascript
\033[38;2;255;82;197mHello
=>
\x1b[38;2;255;82;197mHello
```

At last, it output the code 
```javascript
console.log("[\x1B[38;2;0;191;255mhello\x1B[0m]")
```
